test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -156,6 +157,40 @@ public void AutoFitRankingTest()
                 Assert.True(col.Name == expectedOutputNames[col.Index]);
         }
 
+        [LightGBMFact]
+        public void AutoFitRankingCVTest()
+        {
+            string labelColumnName = "Label";
+            string groupIdColumnName = "GroupIdCustom";
+            string featuresColumnVectorNameA = "FeatureVectorA";
+            string featuresColumnVectorNameB = "FeatureVectorB";
+            uint numFolds = 3;
+
+            var mlContext = new MLContext(1);
+            var reader = new TextLoader(mlContext, GetLoaderArgsRank(labelColumnName, groupIdColumnName,
+                featuresColumnVectorNameA, featuresColumnVectorNameB));
+            var trainDataView = reader.Load(new MultiFileSource(DatasetUtil.GetMLSRDataset()));
+
+            CrossValidationExperimentResult<RankingMetrics> experimentResult = mlContext.Auto()
+                .CreateRankingExperiment(new RankingExperimentSettings() { GroupIdColumnName = groupIdColumnName, MaxExperimentTimeInSeconds = 5 })
```

Comment:
**Review**: When the user uses a custom `GroupIdColumnName`, the user must also specify it in  RankingExperimentSettings(). Is this scenario ok? If so, where would be a good place to document that?

Reply:
So I wouldn't know, personally I don't understand what was the design decision behind letting the user provide a GroupIdColumnName separately on `RankingExperimentSettings` and `ColumnInformation`, as it seems that both should always be the same.

If anything, I'd recommend adding a note on the docs for `RankingExperimentSettings`.

---
In reply to: [451912609](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451912609) [](ancestors = 451912609)

Reply:
After looking more into it, it seems somewhat complicated to redesign these so that the user only needs to provide the custom name for GroupId once.

If anything, I'd suggest adding a `ValidateGroupIdColumnName()` method on the `Execute()` overloads that checks that the GroupIdName matches the one present on the Metrics Agent. And, also sync with ModelBuilder team about the need of setting the custom name twice.

An alternative, would be to redesign these by making the `ColumnInformation.GroupIdColumnName` field internal, and automatically set its value to the one present on the Metrics Agent inside the `Execute()` methods. This way users would only be able to set the `GroupIdColumnName` when creating the experiment, and not able to set the `ColumnInformation.GroupIdColumnName` themselves. Problem with this is that it's probably not very maintainable, as it assumes that the `Execute()` methods are the only ones where an experiment can be executed / the only place where `ColumnInformation.GroupIdColumnName` can be set... if there are any other APIs where this happens (or if any other API is added in the future), then automatically setting the field would also be necessary there, leading to potential maintainability issues. So Idk if I'd go with this other option.

---
In reply to: [451958714](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451958714) [](ancestors = 451958714,451912609)

Reply:
The reason for passing it as an option was suggested on my last PR, so that we didn't have to change the way AutoML evaluates internally.

Reply:
If we stick with needing the user to provide the name twice, then I'd still suggest adding a `ValidateGroupIdColumnName` inside the `Execute()` methods that actually throws if both strings are different.

Not throwing would result on unexpected behavior if the user actually provides different names for the settings and for the column information.

---
In reply to: [452243061](https://github.com/dotnet/machinelearning/pull/5263#discussion_r452243061) [](ancestors = 452243061)

Reply:
The user will not provide the GroupId twice.